language-support/scala/codegen/codegen.bzl

```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 def _dar_to_scala_impl(ctx):
-    codegen_out_dir = ctx.outputs.codegen_out
+    codegen_out_dir = ctx.actions.declare_directory(ctx.label.name + "_codegen_out")
```

Comment:
I think I'm missing something here. Why is this hard coded now?

Reply:
The ```codegen_out``` output was being 'misused' as the output _directory_ of the code generation but as the generated code was being zipped into the ```srcjar_out``` output the ```codegen_out``` was actually a temporary directory and (luckily) not used as a dependency. The change reflects the actual use of the 'output' and allows the ```scala_source_jar``` rule to work correctly for this ```//language-support/scala/examples:quickstart-scala-codegen-lib``` target.

Reply:
👍